src/components/epr/BasicInfo.js

```diff
@@ -0,0 +1,171 @@
+import React from "react";
+import "./BasicInfo.css";
+import { eventRepeatOptions } from "../../services/eventServices";
+import { Button, Container, Row, Col, Form, FormGroup, FormFeedback, Label, Input } from "reactstrap";
+
+export default class BasicInfo extends React.Component  {
+    constructor(props) {
+        super(props);
+
+        //pre-populates dropdown options using enum
+        this.repeatOptions = [];
+        for(const option in eventRepeatOptions) {
+            this.repeatOptions.push(<option>{option}</option>)
+        }
+
+        this.state = {
+            repeatCycle: "None",
+            startAmPm : "AM",
+            endAmPm: "AM",
+            invalid: {
+                name: false,
+                publicDescription: false,
+                privateDescription: false,
+                numParticipants: false,
+                date: false,
+                startHour: false,
+                startMinute: false,
+                endHour: false,
+                endMinute: false
+            }
+        };
+    }
+
+    //Processes changes in the input and sets them in state
+    handleChange = (event) => {
+        const name = event.target.name;
+        const value = event.target.value;
+
+        this.setState({
+            [name] : value
+        });
+    }
+
+    //Does basic check to see if all fields are occupied
+    validate = () => {
+        let isFullyValid = true;
+        let invalidCopy = JSON.parse(JSON.stringify(this.state.invalid));
+
+        for(let field in invalidCopy) {
+            if(!this.state[field]) {
+                invalidCopy[field] = true;
+                isFullyValid = false
+            }
```

Comment:
Could you add an `else` here to set the field to valid if it was previously invalid? It might be confusing if the user fixed one issue but not all of them and they still showed up as invalid